src/Symfony/Component/AssetMapper/ImportMap/ImportMapAuditor.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Symfony\Component\AssetMapper\ImportMap;
+
+use Symfony\Component\AssetMapper\Exception\RuntimeException;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class ImportMapAuditor
+{
+    private const AUDIT_URL = 'https://registry.npmjs.org/-/npm/v1/security/audits';
+
+    private readonly HttpClientInterface $httpClient;
+
+    public function __construct(
+        private readonly string $importMapConfigPath,
+        HttpClientInterface $httpClient = null,
+    ) {
+        $this->httpClient = $httpClient ?? HttpClient::create();
+    }
+
+    public function audit(): array
+    {
+        $packages = $this->getPackagesVersion();
+
+        if ([] === $packages) {
+            return [];
+        }
+
+        $response = $this->httpClient->request('POST', self::AUDIT_URL, [
+            'json' => [
+                'requires' => array_map(fn(): string => '*', $packages),
+                'dependencies' => array_map(fn(string $version): array => ['version' => $version], $packages),
+            ],
+        ]);
+
+        if (200 !== $response->getStatusCode()) {
+            throw new RuntimeException(sprintf('Error %d auditing packages. Response: %s', $response->getStatusCode(), $response->getContent(false)));
+        }
+
+
+        $data = $response->toArray();
+
+        return [
+            'packages' => $packages,
+            'advisories' => $data['advisories'],
+            'summary' => $data['metadata']['vulnerabilities'],
+        ];
+    }
+
+    private function getPackagesVersion(): array
+    {
+        $path = $this->importMapConfigPath;
+        $importMapConfig = is_file($path) ? (static fn () => include $path)() : [];
```

Comment:
I'm leaning towards injecting `ImportMapManager` into this class and making `loadImportMapEntries()` public inside that class. Then we could just call that from here and work with `ImportMapEntry` objects.

Reply:
You're right. Using `ImportMapManager` will avoid breaks if `importmap.php` format changes.